src/components/cl/hier/allgatherv/allgatherv.c

```diff
@@ -0,0 +1,381 @@
+
+    int is_root = nrank == node_root;
+
+    c64 = UCC_COLL_ARGS_COUNT64(&coll_args->args);
+    d64 = UCC_COLL_ARGS_DISPL64(&coll_args->args);
+
+    if (c64 ^ d64) {
+        cl_debug(team->context->lib,
+                 "mixed 64 bit count/displ mode is not supported\n");
+        return UCC_ERR_NOT_SUPPORTED;
+    }
+
+    cl_schedule = ucc_cl_hier_get_schedule(cl_team);
+    if (ucc_unlikely(!cl_schedule)) {
+        return UCC_ERR_NO_MEMORY;
+    }
+    schedule = &cl_schedule->super.super;
+    memcpy(&args, coll_args, sizeof(args)); 
+
+    UCC_CHECK_GOTO(ucc_schedule_init(schedule, &args, team), out, status);
+
+    // Question: What is this ? 
+    if (n_frags > 1) {
+        args.max_frag_count =
+            ucc_buffer_block_count(args.args.src.info.count, n_frags, 0);
+        args.mask |= UCC_BASE_CARGS_MAX_FRAG_COUNT;
+    }
+
+    full_size    = cl_team->sbgps[UCC_HIER_SBGP_FULL].sbgp->group_size;
+    node_size    = cl_team->sbgps[UCC_HIER_SBGP_NODE].sbgp->group_size;
+    leaders_size = cl_team->sbgps[UCC_HIER_SBGP_NODE_LEADERS].sbgp->group_size;
+    elem_size    = c64 ? 8 : 4;
+
+    // if (!cl_team->sbgps[UCC_HIER_SBGP_NODE].sbgp->preserves_order){
+    //     printf("Reordering node ranking is not supported");
+    //     return UCC_ERR_NOT_SUPPORTED;
+    // }
+
+    // Init buffers for collectives arguments
+    size_t scratch_size = elem_size * (node_size * 2 + leaders_size * 2);
+    status = ucc_mc_alloc(&cl_schedule->scratch, scratch_size, UCC_MEMORY_TYPE_HOST);
+    if (ucc_unlikely(UCC_OK != status)) {
+        cl_error(team->context->lib,
+                 "failed to allocate %zd bytes for full counts",
+                 scratch_size);
+        goto out;
+    }
+    gv_rc    = cl_schedule->scratch->addr; /* +node_size */
+    gv_displ = PTR_OFFSET(gv_rc, node_size*elem_size); /* +node_size*/
+    agv_rc   = PTR_OFFSET(gv_displ, node_size*elem_size); /* +leaders_size*/
+    agv_displ   = PTR_OFFSET(agv_rc, leaders_size*elem_size); /* +leaders_size*/
+
+    // Gatherv in the node
+    // src.info.buffer  -> dst.info_v.buffer
+    if (node_size > 1){
+
+        do { // This section need to be rewritten to support both uint32_t and uint64_t once the logic is approved
+            int _i;
+            uint32_t _scount, _displ;
+            
+            _displ = 0;
+            
+            /* For every rank in the node, add his count as is and the displacement 
+            *    to be the running sum of the counts 
+            */
+            for (_i = 0; is_root && _i < node_size; _i++) {
+                ucc_rank_t r = ucc_ep_map_eval((*cl_team->sbgps[UCC_HIER_SBGP_NODE].sbgp).map, _i);
+                _scount = ((uint32_t *)coll_args->args.dst.info_v.counts)[r];
+                ((uint32_t *)gv_rc)[_i] = _scount;
+                ((uint32_t *)gv_displ)[_i] = _displ;
+                
+                _displ += _scount;
+            }
+        } while (0);
+
+        args.args.coll_type = UCC_COLL_TYPE_GATHERV;
```

Comment:
in case of inplace allgatherv args.src.info is ignored, but for gatherv args.src.info is ignored only at root. Need to set args.src.info for non root ranks